dependencies.yaml

```diff
@@ -33,25 +36,42 @@ files:
       - cudatoolkit
       - docs
       - py_version
+  pyproject_build_reqs: # TODO: This name gets ignored for pyproject.toml (or conda meta.yaml)
+    output: pyproject_build
+    includes:
+      - build
+      - cuda-python
+  pyproject_dependencies: # TODO: This name gets ignored for pyproject.toml (or conda meta.yaml)
+    output: pyproject_dependencies
+    includes:
+      - run
+      - cuda-python
+  test: # TODO: Currently using this name as the array key in the TOML file, but that's a bad idea long term.
+    output: pyproject_optional_dependencies
+    includes:
+      - test_python
 channels:
   - rapidsai
   - conda-forge
 dependencies:
   build:
     common:
-      - output_types: [conda, requirements]
+      - output_types: [conda, requirements, pyproject_build]
         packages:
           - cmake>=3.23.1,!=3.25.0
-          - cuda-python>=11.7.1,<12.0
           - cython>=0.29,<0.30
           - ninja
-          - python>=3.8,<3.11
           - scikit-build>=0.13.1
           - tomli
       - output_types: conda
         packages:
           - fmt>=9.1.0,<10
           - spdlog>=1.11.0,<1.12
+          - python>=3.8,<3.11
+      - output_types: pyproject_build
+        packages:
```

Comment:
```suggestion
        packages:
          - cuda-python>=11.7.1,<12.0
```

Reply:
This isn't quite right, with the issue being that cuda-python is both a build-time and a runtime dependency so I removed it from the build list to place it into a separate, reusable list.